addon/event_dispatcher.js

```diff
@@ -0,0 +1,93 @@
+import Ember from "ember";
+import { inject as service } from "@ember/service";
+import { action } from "@ember/object";
+import { getOwner } from "@ember/application";
+import LinkComponent from "@ember/routing/link-component";
+
+export default class HTMLAnchorEventDispatcher extends Ember.EventDispatcher {
+  @service router;
+
+  findAnchorElement(event) {
+    let target = event.target;
+    let element = null;
+
+    do {
+      if (target instanceof HTMLAnchorElement) {
+        element = target;
+        break;
+      }
+
+      target = target.parentNode;
+    } while (target && target.nodeType === 1);
+
+    return element;
+  }
+
+  recognizeAnchorURL(event, element) {
+    if (
+      !(
+        (event.which === undefined || event.which === 1) &&
+        !event.ctrlKey &&
+        !event.metaKey
+      )
+    ) {
+      return null;
+    }
+
+    if (element.getAttribute("target") === "_blank") {
+      return null;
+    }
+
+    if (element.hasAttribute("download")) {
+      return null;
+    }
+
+    if (element.hasAttribute("data-ember-action")) {
+      return null;
+    }
+
+    let view = getOwner(this).lookup("-view-registry:main")[element.id];
+    if (view && view instanceof LinkComponent) {
+      return null;
+    }
+
+    let href = element.getAttribute("href");
+    if (!href || href.indexOf(this.router.rootURL) !== 0) {
+      return null;
+    }
+
+    if (!this.router.recognize(href)) {
+      return null;
+    }
+
+    return href.replace(this.router.rootURL, '/');
```

Comment:
Removing `rootURL` is required, verifies https://github.com/intercom/ember-href-to/commit/469d10fc5f5a00b26bee55417f4635d8626e6395.